Makefile

```diff
@@ -1,11 +1,18 @@
-
 GOCMD=$(shell which go)
 DOCKER=$(shell which docker)
 GOBINPATH=$(shell $(GOCMD) env GOPATH)
 NPM=$(shell which npm)
 STATIK=$(GOBINPATH)/bin/statik
 GOLANGCILINT_VERSION=v1.25.1
 
+# check for mandatory dependencies - binaries can be overridden.
```

Comment:
general question - why do we perform this check always. If I use the makefile to generate docs I am not required to have all these tools.
can you condition the check to the stop or add another step to check requirements?